go-controller/pkg/util/util.go

```diff
@@ -188,3 +188,30 @@ func HashForOVN(s string) string {
 	return fmt.Sprintf("a%s", hashString)
 
 }
+
+// UpdateIPsSlice will search for values of v in the slice and update it with n values of same IP family
+func UpdateIPsSlice(s, oldIPs, newIPs []string) []string {
```

Comment:
The comment is not clear about what the function does, or at least I´m not getting what `v` and `n` is

Reply:
also,due to the complexity we'll need an unit test

Reply:
https://play.golang.org/p/b-xyepqieM2

```go
func TestUpdateIPsSlice(t *testing.T) {
	var tests = []struct {
		name              string
		s, oldIPs, newIPs []string
		want              []string
	}{
		{
			"test1",
			[]string{"192.168.1.1", "10.0.0.1", "127.0.0.2"},
			[]string{"192.168.1.0/24"},
			[]string{"192.168.2.0/24"},
			[]string{},
		},
		{
			"test2",
			[]string{"192.168.1.1", "10.0.0.1", "127.0.0.2"},
			[]string{"192.168.5.0/24"},
			[]string{"192.168.2.0/24"},
			[]string{},
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {

			ans := UpdateIPsSlice(tt.s, tt.oldIPs, tt.newIPs)
			if !reflect.DeepEqual(ans, tt.want) {
				t.Errorf("got %v, want %v", ans, tt.want)
			}

		})
	}
}
```

Reply:
thanks I'll add a unit test